samples/service_manager_esxtop_in_vc.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+import atexit
+import argparse
+import getpass
+import ssl
+
+from pyVmomi import vim
+from pyVim.connect import SmartConnect, Disconnect
+
+"""
+Example of fonnecting to the ESXTOP service provided 
+by vCenter Server's Service Manager
+"""
+
+__author__ = 'William Lam'
+
+
+def GetArgs():
+    """
+    Supports the command-line arguments listed below.
+    """
+    parser = argparse.ArgumentParser(
+        description='ESXTOP metric collection via vCenter Server')
+    parser.add_argument('-s', '--host', required=True, action='store',
+                        help='Remote host to connect to')
+    parser.add_argument('-o', '--port', type=int, default=443, action='store',
+                        help='Port to connect on')
+    parser.add_argument('-u', '--user', required=True, action='store',
+                        help='User name to use when connecting to host')
+    parser.add_argument('-p', '--password', required=False, action='store',
+                        help='Password to use when connecting to host')
+    parser.add_argument('-e', '--esxi', required=False, action='store',
+                        help='ESXi hostname/IP managed by vCenter Server')
+    args = parser.parse_args()
+    return args
+
+
+# Start program
+def main():
+    args = GetArgs()
+    if args.password:
+        password = args.password
+    else:
+        password = getpass.getpass(
+            prompt='Enter password for host %s and '
+                   'user %s: ' % (args.host, args.user))
+
+    si = SmartConnect(host=args.host,
+                      user=args.user,
+                      pwd=password,
+                      port=int(args.port),
+                      sslContext=ssl._create_unverified_context())
+
+    atexit.register(Disconnect, si)
+
+    # "vmware.host." prefix is required when connecting to VC
+    location = "vmware.host." + args.esxi
+
+    services = si.content.serviceManager.QueryServiceList(
+        location=[location])
```

Comment:
We actually only have 2 hardcoded ones, so I figure it would be simple to just keep it as is :)